MockKit/MockCGMManager.swift

```diff
@@ -45,23 +57,98 @@ public struct MockCGMState: SensorDisplayable {
         }
     }
     
+    private var cgmLowerLimitValue: Double
+    
+    public var cgmLowerLimit: HKQuantity {
+        get {
+            return HKQuantity.init(unit: HKUnit.milligramsPerDeciliter, doubleValue: cgmLowerLimitValue)
+        }
+        set {
+            cgmLowerLimitValue = newValue.doubleValue(for: HKUnit.milligramsPerDeciliter)
+        }
+    }
+    
+    private var cgmUpperLimitValue: Double
+    
+    public var cgmUpperLimit: HKQuantity {
+        get {
+            return HKQuantity.init(unit: HKUnit.milligramsPerDeciliter, doubleValue: cgmUpperLimitValue)
+        }
+        set {
+            cgmUpperLimitValue = newValue.doubleValue(for: HKUnit.milligramsPerDeciliter)
+        }
+    }
+    
     public var cgmStatusHighlight: MockCGMStatusHighlight?
     
-    public var cgmStatusProgress: MockCGMStatusProgress?
+    public var cgmLifecycleProgress: MockCGMLifecycleProgress? {
+        didSet {
+            if cgmLifecycleProgress != oldValue {
+                setProgressColor()
+            }
+        }
+    }
+    
+    public var progressWarningThresholdPercentValue: Double? {
+        didSet {
+            if progressWarningThresholdPercentValue != oldValue {
+                setProgressColor()
+            }
+        }
+    }
+    
+    public var progressCriticalThresholdPercentValue: Double? {
+        didSet {
+            if progressCriticalThresholdPercentValue != oldValue {
+                setProgressColor()
+            }
+        }
+    }
+    
+    private mutating func setProgressColor() {
+        guard cgmLifecycleProgress != nil else {
```

Comment:
[nit] Consider `guard let cgmLifecycleProgress = cgmLifecycleProgress else {` and getting rid of the forced unwrapping below.